### ISSUE TITLE
Add Malayalam script

### ISSUE THREAD

**Issue by kalapi** [googlefonts/googlesans#555]
Hello @madig, @MariannaPaszkowska, and @chrissimpkins 

I'm attaching the finalised Google Sans Malayalam v1.000 package here. 

Here are some false positives re. GDEF:

```
The following mark characters could be in the GDEF:
mark glyph class: uni0D41 (U+0D41), uni0D42 (U+0D42), uni0D43 (U+0D43), and uni0D44 (U+0D44)
```

Another thing to note–we're going to be using the localised international numerals from the Tamil glyph-set, and to that end i've included a `locl` lookup pointing the default figs to ones Malayalam should use. I can test the functionality after you have compiled the first round. 

As always, please let me know if you have any questions or comments–Thanks!

[2022-11-15_GoogleSansMalayalam_V1.000_BETA01.zip](https://github.com/googlefonts/googlesans/files/10011168/2022-11-15_GoogleSansMalayalam_V1.000_BETA01.zip)

**Comment by MariannaPaszkowska** [comment]
Thank you, @kalapi!
@belluzj @madig @Hoolean @RickyDaMa, as discussed, we need to postpone the review of Malayalam. We will reach out to schedule it. 
CC @chrissimpkins

**Comment by madig** [comment]
@kalapi I suppose the Tamil locl lookups are the same as the actual Tamil ones, so I'm just referring to those directly in the import. Builds are underway at https://github.com/googlefonts/googlesans/actions/runs/4796497188, can you please have a cursory look once the fonts there are done? Then I can pass them on to internal review.

**Comment by kalapi** [comment]
> @kalapi I suppose the Tamil locl lookups are the same as the actual Tamil ones [...]

I didn't understand this? Did you mean that the Malayalam locl lookups are the same as the Tamil ones?

**Comment by madig** [comment]
Yes, so instead of redefining them, I reuse them.

https://github.com/googlefonts/googlesans/blob/e8f076e31c9f57838f179d0a1ddeac72889db42e/source/GoogleSans/malayalam.fea#L30-L31

**Comment by kalapi** [comment]
Ah got it! I realised I added the first one by mistake. It should be:

```script mlm2;

lookup MALM_dflt_locl_punctuation {
	sub period by period.loclMALM;
	sub comma by comma.loclMALM;
	sub colon by colon.loclMALM;
	sub semicolon by semicolon.loclMALM;
	sub ellipsis by ellipsis.loclMALM;
	sub exclam by exclam.loclMALM;
	sub question by question.loclMALM;
	sub asterisk by asterisk.loclMALM;
	sub parenleft by parenleft.loclMALM;
	sub parenright by parenright.loclMALM;
	sub braceleft by braceleft.loclMALM;
	sub braceright by braceright.loclMALM;
	sub bracketleft by bracketleft.loclMALM;
	sub bracketright by bracketright.loclMALM;
	sub quotedblleft by quotedblleft.loclMALM;
	sub quotedblright by quotedblright.loclMALM;
	sub quoteleft by quoteleft.loclMALM;
	sub quoteright by quoteright.loclMALM;
} MALM_dflt_locl_punctuation;

lookup MALM_dflt_locl_malayalam_danda {
	sub danda-deva by danda-deva.loclMALM;
	sub dbldanda-deva by dbldanda-deva.loclMALM;
} MALM_dflt_locl_malayalam_danda;

lookup tml2_dflt_locl_localised_numerals;
```

We're only reusing the numerals from the Tamil. Sorry for the confusion!

**Comment by kalapi** [comment]
Also, i've checked the binaries in the link you provided and they look good!

**Comment by kalapi** [comment]
Hi @MariannaPaszkowska & @chrissimpkins 

Could you please add Joana Ranito (https://github.com/joanaranito) to this repo please?

Many thanks in advance!

**Comment by chrissimpkins** [comment]
> Hi @MariannaPaszkowska & @chrissimpkins 
> 
> 
> 
> Could you please add Joana Ranito (https://github.com/joanaranito) to this repo please?
> 
> 
> 
> Many thanks in advance!

done!

**Comment by MariannaPaszkowska** [comment]
@kalapi @joanaranito, hi there, just wanted to check in and see how things are going with the sources. Any update on when we can expect the delivery? Thanks for your help!

**Comment by joanaranito** [comment]
Hi @MariannaPaszkowska, I'm aiming to upload the sources by the end of today! Just running some final checks on the fonts :)

**Comment by joanaranito** [comment]
Hi @MariannaPaszkowska @madig @chrissimpkins ! Here's the updated package for Google Sans Malayalam v1.000.
We've fixed the issues pointed out by the QA, and generally improved the consistency of the inktraps. There are some slight changes to spacing due to updated metrics keys.

I've attached the new package below. Let me know if you need anything else or have any comments. 

Thanks!

[2023-06-01_GoogleSansMalayalam_V1-2.000_BETA02.zip](https://github.com/googlefonts/googlesans/files/11627259/2023-06-01_GoogleSansMalayalam_V1-2.000_BETA02.zip)

**Comment by chrissimpkins** [comment]
Thank you very much Joana!

**Comment by madig** [comment]
@joanaranito Hi! Did you modify the locl feature?

**Comment by joanaranito** [comment]
Hi @madig ! 

No, I didn't modify the locl feature in the sources. Rereading the whole thread I realize I probably should have removed the lookup `tml2_dflt_locl_localised_punctuations`, is that right? If so, I can do that and re-upload the sources.

I couldn't figure out why the `MALM_dflt_locl_punctuation` was not working in the build Hanna was testing (assuming it was a different build than the one shared above), would you be able to share that so we can have a look?

**Comment by madig** [comment]
Sure, see the internal doc

**Comment by joanaranito** [comment]
@madig If it's the Malayalam_Import.zip, I don't have access :) (I don't have access to any of the links in the doc except to the link to this issue)

**Comment by madig** [comment]
(Passed by mail)

@joanaranito Hanna has a question about the locl feature, please see the doc :)

**Comment by joanaranito** [comment]
Hey @madig I replied to the question in the doc :)

**Comment by madig** [comment]
Thanks, Hanna will be back next week to have a look.

**Comment by joanaranito** [comment]
Hey @MariannaPaszkowska @madig @chrissimpkins, here are the updated sources and fonts for the Malayalam!

I added the opentype substitution for the v_va-malayalam and y_ya-malayalam, and added the virama anchor in n_rra-malayalam to fix the clash with the virama.

I also removed the `MALM_dflt_locl_punctuation` lookup from the `locl` feature.

The remaining shaping differences that Marianna reported in issue #580 are expected. I'm in the process of ticking them off☺️ 

Let me know if you need anything else or have any questions!

[2023-06-21_GoogleSansMalayalam_V1-2.000_BETA03.zip](https://github.com/googlefonts/googlesans/files/11821755/2023-06-21_GoogleSansMalayalam_V1-2.000_BETA03.zip)

**Comment by madig** [comment]
@joanaranito hi! did you remove `lookup tml2_dflt_locl_localised_numerals;` instead?

**Comment by joanaranito** [comment]
Oh, @madig sorry, you're right! I copied the wrong lookup name! I didn't remove `MALM_dflt_locl_punctuation`, I removed `tml2_dflt_locl_localised_numerals`

Sorry about the communication mistake :)

**Comment by madig** [comment]
Thanks. Can you please confirm that the VF builds in https://github.com/googlefonts/googlesans/suites/13826983037/artifacts/767529144 contain your changes?

**Comment by joanaranito** [comment]
Sure, looking into them now!

**Comment by joanaranito** [comment]
Hey @madig, I can confirm that the VF builds you shared contain our changes :)

**Comment by madig** [comment]
Thanks :) Oh btw, since Hanna is on leave currently, why did you remove the lookup again?

**Comment by joanaranito** [comment]
Oh, @madig  I'm sorry, I thought I was supposed to! 

But now that I'm re-reading the thread I realise I wasn't supposed to remove it, I'm really sorry! 🤦 Should I re-add and send the sources again?

**Comment by joanaranito** [comment]
I somehow misunderstood @kalapi's comment: https://github.com/googlefonts/googlesans/issues/555#issuecomment-1522197941

**Comment by madig** [comment]
I'm confused now :D I did not talk about this with Hanna, I just assumed that's what you two settled on. I'm just the janitor taking sources and merging them, I'll leave the thinking to you and @kalapi 😬

**Comment by joanaranito** [comment]
Sorry, @madig, what I mean is that I misunderstood the thread and removed the lookup, even though I wasn't supposed to. My question was if I should add the lookup back to the sources and re-send the sources, or if it's easier for you to add it on your end.

**Comment by madig** [comment]
Oh, ok. I'll add it back.

**Comment by madig** [comment]
Actually. @joanaranito can you put it in and also generate new fonts for us to check against?

**Comment by joanaranito** [comment]
Of course, will do it now!

**Comment by joanaranito** [comment]
@madig Here are the updated sources and TTFs. Sorry again for the confusion!

[2023-06-28_GoogleSansMalayalam_V1-2.000_BETA04.zip](https://github.com/googlefonts/googlesans/files/11893238/2023-06-28_GoogleSansMalayalam_V1-2.000_BETA04.zip)

**Comment by madig** [comment]
Thanks; I had Jonny look at the beta 3 package and told him to ignore the locl stuff. Does your beta 4 contain any other change? If not, once you dismiss his question in the doc, we have green light.

**Comment by joanaranito** [comment]
No, no other changes other than the locl feature. I just replied to Jonny on the doc. The issue he flagged was one of the expected behaviours flagged in issue #580, and something @kalapi and I discussed with @MariannaPaszkowska in our last catch-up :)

**Comment by MariannaPaszkowska** [comment]
Closed with #575 Thank you so much, great work everyone!
